sucury.py

```diff
@@ -407,18 +422,19 @@ def play():
             # Key pressed
             if event.type == pygame.KEYDOWN:
                 if game_on:
-                    if event.key == pygame.K_DOWN  and snake.ymov == 0:    # Down arrow:  move down
-                        snake.ymov = 1
-                        snake.xmov = 0
-                    elif event.key == pygame.K_UP  and snake.ymov == 0:    # Up arrow:    move up
-                        snake.ymov = -1
-                        snake.xmov = 0
+                    new_direction = None
+                    if event.key == pygame.K_DOWN and snake.ymov == 0:  # Down arrow: move down
```

Comment:
You can add the possibility that the snake could move upwards and then in a direction parallel to the one it was already moving in, for example:

if event.key == pygame.K_RIGHT and skane.ymov ==0 and len(snake.movemente_queue) == 1:
      new_direction = (1 ,0 )
      
 
the same logic for other cases (right, left, up, down)